### ISSUE TITLE
Can not load Genesis ROM > 10MB

### ISSUE THREAD

**Issue by TheRetroSnake** [TASVideos/BizHawk#543]
So, I noticed while testing with Bizhawk, that you can not load ROM image > 10MB. While this is much better than most emulators (*cough* only supports 6MB *cough*), this is still incorrect when SSFII mapper is in use. In fact, while using it, up to 32MB ROM's should work just fine, as it is the hard limit of said mapper, according to HW information we have, and this document http://emu-docs.org/Genesis/ssf2.txt
I would like the emulator to have support for ROMs > 10MB in size. It would likely to be logical to load first 4 or 8 MB to ROM section, and then rely on the source program to map different portions of the ROM to different banks.

**Comment by zeromus** [comment]
Your understanding of bank switching is pretty garbled. Nonetheless, it seems youre right, the docs say up to 16MB is swappable. However, this is a bug for GPGX upstream. If GPGX has fixed it then let us know and we'll investigate updating our GPGX. Supplying an actual test rom would be helpful.

**Comment by zeromus** [comment]
Looks like GPGX only enables the SF2 mapper with varying parameters via a whitelist. if you're trying to make an oversized homebrew using this mapper, then youll need to find some genesis emulator programmers to get a consensus for a new "board type". I would suggest (pretty ignorantly, as i dont know much about genesis or GPGX) for it to be ROMType "SF" and product "64xx" where xx corresponds to the existing functionality you want and 64 is the new part that says it should support all 64 banks

**Comment by TheRetroSnake** [comment]
Actually 32MB are swappable. 512KB*64 banks = 32MB. Also SSFII mapper works for example on Mega Everdrive, pretty much as it did with SSFII (with few additions). Also SSFII is fairly good system as is, but the issue is emulator compatability. One other issue in Bizhawk is that it expects the SSFII header to enable swapping, else it would not work (at least according to my tests) which should really not be the case.
Either way I'll try to see about GPGX and if it can be fixed in any regards. Also have a ROM which is 32MB and will print each bank after another. http://natsumi.noip.me/d/misc/a.zip/

**Comment by hegyak** [comment]
Can you confirm your Dump is a good ROM?  I just checked my No-Intro dump and the game appears to be loaded/working?  The demo mode is playing and it appears to be correct.  Or is the issue somewhere else?

**Comment by zeromus** [comment]
He's wanting to run a hypothetical rom, not SF2.

As I said, the GPGX core that bizhawk uses employs a whitelist, yes, by checking the header. How else are you supposed to know which mapper to use?

**Comment by TheRetroSnake** [comment]
From my tests Regen and Fusion activate the mapper when I actually try to map the banks. That being said, any header setup should work. For example, Sonic VR (https://info.sonicretro.org/Sonic_VR) uses bankswitching to surpass the 4MB limit, and both Regen and Fusion work with it just fine, yet Bizhawk would not work so well considering the whitelist. I am not aware if some other application use 0xA130F3-0xA130FF range, but if not many, then it should be reasonably easy to just assume any ROM can bankswitch, as many wont try to change banks anyway.

**Comment by zeromus** [comment]
It's still a project for GPGX

**Comment by TheRetroSnake** [comment]
ok, I talked with the GPGX devs, and here is what was said specifically: https://github.com/ekeeke/Genesis-Plus-GX/issues/67
tl;dr changing MAXROMSIZE in loadrom.h to 32MB over the regular value should be harmless and since its PC application it shouldn't cause any issues with vast majority of users

**Comment by zeromus** [comment]
looks like it's working okay now, GPGX already handled it once the rom size restrictions were lifted
